go/engine/login.go

```diff
@@ -82,17 +83,50 @@ func (e *Login) Run(ctx *Context) error {
 
 	e.G().Log.Debug("LoginCurrentDevice error: %s (continuing with device provisioning...)", err)
 
-	// this device needs to be provisioned:
+	// this device needs to be provisioned
+
+	// clear out any existing session:
+	e.G().Logout()
+
+	// transaction around config file
+	tx, err := e.G().Env.GetConfigWriter().BeginTransaction()
+	if err != nil {
+		return err
+	}
+
+	// From this point on, if there's an error, we abort the
+	// transaction.
+	defer func() {
+		if tx != nil {
+			tx.Abort()
+		}
+	}()
+
+	// run the username engine to load a user
+	ueng := NewLoginUsername(e.G(), e.username)
```

Comment:
This engine is also confusingly named.  It's not actually logging in the user...